modules/service/src/main/java/com/devonfw/module/service/common/api/client/SyncServiceClient.java

```diff
@@ -0,0 +1,77 @@
+package com.devonfw.module.service.common.api.client;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+/**
+ * An {@link SyncServiceClient} wraps a {@link #get() service client} allowing to do {@link #call(Object, Consumer)
+ * synchronous calls}. Only {@link ServiceClientFactory} is thread-safe but not instances of this interface so please do
+ * not share clients between threads. The following code is a simple example how to do an synchronous service
+ * invocation:
+ *
+ * <pre>
+ * {@link ServiceClientFactory} factory = getServiceClientFactory();
+ * {@link SyncServiceClient}{@literal <MyService>} client = factory.{@link ServiceClientFactory#createSync(Class) createSync}(MyService.class);
+ * client.{@link #call(Object, Consumer) call}(client.{@link #get()}.doSomethingOperation(args), r -> {
+ *   handleSyncResult(r);
+ * });
+ * </pre>
+ *
+ * Please note that the {@link #get() actual service client} is just a stub recording the desired service operation to
+ * call synchronously. It will only return a dummy result (typically {@code null} and for primitive types an according
+ * default like {@code 0} or {@code false}). This approach allows the {@link #call(Object, Consumer) call} method to be
+ * type-safe ensuring that the {@link Consumer} callback fits to the result type of the invoked operation.<br>
+ * Therefore ensure that you always call exactly one service operation on the {@link #get() service client} immediately
+ * followed by a {@link #call(Object, Consumer) call} method invocation.
+ *
+ * @param <S> type of the {@link #get() service client}.
+ * @since 2021.08.003
+ * @see ServiceClientFactory#createSync(Class)
+ */
+public interface SyncServiceClient<S> {
```

Comment:
OK. It seems there is a missunderstanding here.
The synchronous access using HTTP should behave exactly like the API we already have existing.
Only the a new implementation option should be added that can be used instead without `cxf-client` dependencies...

IMHO for synchronous invocation this new API seems overcomplicated. I do not have to provide `Consumer` as callback when I can call the code directly.

Reply:
See existing documentation:
https://github.com/devonfw/devon4j/blob/master/documentation/guide-service-client.asciidoc#simple-usage

```
 private void callSynchronous(MyArgs myArgs) {
    MyService myService = this.serviceClientFactory.create(MyService.class);
    // call of service over the wire, synchronously blocking until result is received or error occurred
    MyResult myResult = myService.myMethod(myArgs);
    handleResult(myResult);
  }
```

So when replacing dependency `devon4j-starter-cxf-client-rest` with `devon4j-starter-http-client-rest-sync` the same code pattern should work except that under the hood no more CXF but the HttpClient is used.